libgalois/src/PropertyGraph.cpp

```diff
@@ -67,17 +67,37 @@ MapTopology(const tsuba::FileView& file_view) {
 
   auto* out_dests = reinterpret_cast<uint32_t*>(out_indices + num_nodes);
 
-  auto indices_buffer = std::make_shared<arrow::MutableBuffer>(
-      reinterpret_cast<uint8_t*>(out_indices), num_nodes);
+  auto adj_arr = std::make_unique<katana::LargeArray<uint64_t> >();
+  auto dst_arr = std::make_unique<katana::LargeArray<uint32_t> >();
 
-  auto dests_buffer = std::make_shared<arrow::MutableBuffer>(
-      reinterpret_cast<uint8_t*>(out_dests), num_edges);
+  adj_arr->allocateBlocked(num_nodes);
+  dst_arr->allocateBlocked(num_edges);
 
+  katana::ParallelSTL::copy(out_indices, out_indices + num_nodes, adj_arr->begin());
+  katana::ParallelSTL::copy(out_dests, out_dests + num_edges, dst_arr->begin());
+
+  for (const auto& n: *dst_arr) {
+    KATANA_LOG_ASSERT(n < num_nodes);
+  }
+
+  for (const auto& e: *adj_arr) {
```

Comment:
`do_all` for both these loops if you want to keep them in the merged code.

Reply:
Good point. I'll remove these for now. They were there to unfold the mysteries of previous code :)